### DISCUSSION TITLE
Using rate limiter with before_request

### DISCUSSION THREAD

**Discussion by andrii-korotkov-verkada** [alisaifee/flask-limiter#442] [Q&A]
Hey. I want to specify a configurable rate limit per various endpoints, but don't want to decorate every endpoint with limit decorator and still want to keep using @app.route to register endpoints and not writing a wrapper decorator.
Using a limiter in function decorate with before_request seems ideal, but I haven't found a straightforward way to do so. 
Should I do something like this in a function with before_request decorator or is it a dumb idea?
```
with limiter.limit(...params...):
    pass
```
Ideally, I'm looking for something like `limiter.apply_limit`, a standalone function, not decorator.
Can I achieve the goal with specifying callables in `default_limits`?
Thank you.

**Comment by alisaifee** [comment]
`default_limits` is meant to cover this use case where the limits you configure there are applied to all routes in your application. If you want to get more specific and have Blueprints - you can also apply a limit to the entire blueprint.

**Reply by andrii-korotkov-verkada** [reply]
`default_limits` allows to pass a callable for computing the limit. Just to check, is this callable evaluated every time the limit needs to be applied, or is it compute only in the beginning?
